### PR TITLE
IB: use mlx5dv.h 

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/uct/ib/mlx5/ib_mlx5_dv.c [hunk 1]
```diff
@@ -0,0 +1,162 @@
+#include "ib_mlx5.h"
```
Comment (miked-mellanox) [inline src/uct/ib/mlx5/ib_mlx5_dv.c]: copyright & license header

File: src/uct/ib/mlx5/ib_mlx5_dv.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
+#ifndef UCT_IB_MLX5_DV_H_
```
Comment (miked-mellanox) [inline src/uct/ib/mlx5/ib_mlx5_dv.h]: copyr + lic

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7173/ for details (Mellanox internal link).

File: config/m4/ib.m4 [hunk 3]
```diff
@@ -58,10 +58,10 @@ AC_ARG_WITH([cm],
 #
 # mlx5 bare-metal support
 #
-AC_ARG_WITH([mlx5-hw],
-            [AC_HELP_STRING([--with-mlx5-hw], [Compile with mlx5 bare-metal support])],
+AC_ARG_WITH([mlx5-dv],
+            [AC_HELP_STRING([--with-mlx5-dv], [Compile with mlx5 DV support])],
```
Comment (shamisp) [inline config/m4/ib.m4]: DV is very cryptic name. Let's be more specific what this option provides

Comment (yosefe) [reply config/m4/ib.m4]: @Artemy-Mellanox pls update help message to "Compile with mlx5 direct-verbs support"

Comment (shamisp) [reply config/m4/ib.m4]: I would even expand on this. "Direct Verbs (DV) support provides additional acceleration capabilities that are not available in a regular mode." or something along this lines.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7186/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7207/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
replaced by #2695

**Pull request closed without merging**